### PR TITLE
Document CGet* instructions

### PR DESCRIPTION
**Pull request by arichardson**
These descriptions are based on the existing MIPS prose (converted to
markdown) with a few minor consistency tweaks.

### CODE REVIEW

File: src/cheri_insts.sail [hunk 1]
```diff
@@ -44,20 +44,36 @@ union clause ast = CGetOffset : (regidx, regidx)
...
 {
   let capVal = C(cs1);
   X(rd) = EXTZ(getCapFlags(capVal));
   RETIRE_SUCCESS
 }
 
+/*!
+ * Integer register *rd* is set equal to the sign-extended **otype** field
+ * of capability register *cs1*.
+ *
```
Comment (jrtc27) [inline src/cheri_insts.sail:75]:
```suggestion
```

File: src/cheri_insts.sail [hunk 2]
```diff
@@ -67,20 +83,41 @@ function clause execute (CGetType(rd, cs1)) =
...
+ * inside math mode.
+ */
+/*!
+ * Integer register *rd* is set equal to the **length** field of capability
+ * register *cs1*.
+ *
+ * ## Notes
+ * Due to the compressed representation of capabilities, the actual length
+ * of capabilities can be $2^{[{xlen}][xlen]}$; [CGetLen] will return the maximum
+ * value of $2^{[{xlen}][xlen]}-1$ in this case.
```
Comment (jrtc27) [inline src/cheri_insts.sail:119]: I'm unsure how to deal with notes. Some of them make sense to keep in Sail, but some of the MIPS ones refer to uarch things and don't make so much sense there.

File: src/cheri_insts.sail [hunk 3]
```diff
@@ -89,20 +126,33 @@ function clause execute (CGetLen(rd, cs1)) =
...
+ */
 function clause execute (CGetSealed(rd, cs1)) =
 {
   let capVal = C(cs1);
   X(rd) = EXTZ(bool_to_bits(capVal.sealed));
   RETIRE_SUCCESS
 }
 
+/*!
+ * Integer register *rd* is set equal to the **cursor** field (the address) of
```
Comment (jrtc27) [inline src/cheri_insts.sail:153]: The field is called address in RISC-V.

File: src/cheri_insts.sail [hunk 4]
```diff
@@ -89,20 +126,33 @@ function clause execute (CGetLen(rd, cs1)) =
   RETIRE_SUCCESS
 }
 
+/*!
+ * The low bit of integer register *rd* is set to the **tag** value of *cs1*.
```
Comment (jrtc27) [inline src/cheri_insts.sail:130]: "tag value of cs1" reads strangely; use field instead like the others?

File: src/cheri_insts.sail (hunk 2 above)
Comment (arichardson) [reply src/cheri_insts.sail:119]: I dropped some of the notes that don't make sense for RISC-V, but I felt like this one might make sense here?

File: src/cheri_insts.sail (hunk 4 above)
Comment (arichardson) [reply src/cheri_insts.sail:130]: Sounds good, will update the MIPS description too.

File: src/cheri_insts.sail (hunk 3 above)
Comment (arichardson) [reply src/cheri_insts.sail:153]: We still use `getCapCursor`, so until that is changed we should maybe match the MIPS description?

Comment (jrtc27) [reply src/cheri_insts.sail:152]: Those 3 instances should just be replaced with `cs2.address` like everywhere else and the function removed.

File: src/cheri_insts.sail (hunk 2 above)
Comment (jrtc27) [reply src/cheri_insts.sail:118]: The issue though is what to do if we decide we do have uarch-related notes we want to include.

Comment (arichardson) [reply src/cheri_insts.sail:118]: How about: uarch notes that affect the architectural instruction semantics go into the sail, everything else into the uarch chapter of the ISA ref?

File: src/cheri_insts.sail [hunk 5]
```diff
@@ -44,20 +44,35 @@ union clause ast = CGetOffset : (regidx, regidx)
...
 function clause execute (CGetFlags(rd, cs1)) =
 {
   let capVal = C(cs1);
   X(rd) = EXTZ(getCapFlags(capVal));
   RETIRE_SUCCESS
 }
 
+/*!
+ * Integer register *rd* is set equal to the sign-extended **otype** field
+ * of capability register *cs1*.
```
Comment (jrtc27) [inline src/cheri_insts.sail:74]: That's not quite true; normal otypes are positive, reserved otypes are negative, and the split is asymmetric.

Comment (jrtc27) [reply src/cheri_insts.sail:74]: And arguably like bounds the otype is architecturally an XLEN-bit value with restricted values due to architectural compression...

Comment (arichardson) [reply src/cheri_insts.sail:74]: MIPS has this:
```\subsection*{Notes}

\begin{itemize}
\item
If the capability is unsealed, a value of -1 is returned.
For hardware-interpreted \cotype{}s (see \cref{tab:archotypes}) a sign-extended
(negative) value is returned; for software-defined \cotype{}s
(see \cref{sec:model-sealedcapabilities}) \insnmipsref{CGetType} returns
a zero-extended value.
\end{itemize}
```

File: src/cheri_insts.sail (hunk 2 above)
Comment (jrtc27) [reply src/cheri_insts.sail:118]: That seems sensible; we'll see whether there are any notes that don't fall into either category nicely.

File: src/cheri_insts.sail [hunk 6]
```diff
@@ -67,20 +82,41 @@ function clause execute (CGetType(rd, cs1)) =
...
 
+/*
+ * Note: We have to use [{xlen}][xlen] instead of [xlen] to avoid \lstinline{}
+ * inside math mode.
+ */
+/*!
+ * Integer register *rd* is set equal to the **length** field of capability
+ * register *cs1*.
+ *
+ * ## Notes
```
Comment (jrtc27) [inline src/cheri_insts.sail:115]:
```suggestion
 * ## Notes
 *
```

File: src/cheri_insts.sail (hunk 5 above)
Comment (arichardson) [reply src/cheri_insts.sail:74]: I could just drop the sign-extended and people can look at the sail to figure out the value?

Comment (jrtc27) [reply src/cheri_insts.sail:74]: Yeah that might be easiest, though we could add something reminding people that software-defined otypes are positive and architecturally-reserved otypes are negative.

Comment (jrtc27) [reply src/cheri_insts.sail:74]: Uh, under Notes, that is.

**Pull request merged**